tests/test_cli_compile.py

```diff
@@ -3001,6 +3001,40 @@ def test_cli_compile_strip_extras(runner, make_package, make_sdist, tmpdir):
     assert "[more]" not in out.stderr
 
 
+def test_cli_compile_all_extras_with_multiple_packages(
+    runner, make_package, make_sdist, tmpdir
+):
+    """
+    Assures that ``--all-extras`` works when multiple sources are specified.
+    """
+    test_package_1 = make_package(
+        "test_package_1",
+        version="0.1",
+        extras_require={"more": []},
+    )
+    test_package_2 = make_package(
+        "test_package_2",
+        version="0.1",
+        extras_require={"more": []},
+    )
+
+    out = runner.invoke(
+        cli,
+        [
+            "--all-extras",
+            "--output-file",
+            "requirements.txt",
+            str(test_package_1 / "setup.py"),
+            str(test_package_2 / "setup.py"),
+        ],
+    )
+
+    assert out.exit_code == 0, out
+    assert "--all-extras" in out.stderr
+    assert f"test_package_1{os.path.sep}0.1{os.path.sep}setup.py" in out.stderr
+    assert f"test_package_2{os.path.sep}0.1{os.path.sep}setup.py" in out.stderr
```

Comment:
I think you overlooked @webknjaz's refactoring suggestion:

```suggestion
    test_package_1_setup_py =  f"test_package_1{os.path.sep}0.1{os.path.sep}setup.py"
    test_package_2_setup_py =  f"test_package_2{os.path.sep}0.1{os.path.sep}setup.py"

    out = runner.invoke(
        cli,
        [
            "--all-extras",
            "--output-file",
            "requirements.txt",
            test_package_1_setup_py,
            test_package_2_setup_py,
        ],
    )

    assert out.exit_code == 0, out
    assert "--all-extras" in out.stderr
    assert test_package_1_setup_py in out.stderr
    assert test_package_2_setup_py in out.stderr
```

Reply:
@chrysle only I wanted the pathlib+str for making those vars. OTOH the current state is acceptable, I guess.